app/services/withdraw_application.rb

```diff
@@ -0,0 +1,19 @@
+class WithdrawApplication
+  def initialize(application_choice:)
+    @application_choice = application_choice
+  end
+
+  def save!
+    ActiveRecord::Base.transaction do
+      ApplicationStateChange.new(application_choice).withdraw!
+      application_choice.update(withdrawn_at: Time.zone.now)
```

Comment:
If we're not checking the return value:

```suggestion
      application_choice.update!(withdrawn_at: Time.zone.now)
```